### PR TITLE
cudf now leverages rapids-cmake to reduce CMake boilerplate

### PR DESCRIPTION
**Pull request by robertmaynard**
rapids-cmake providies features such as dependency tracking, pre-configured CPM dependencies, and CUDA architecture detction.

Using those features allows cudf to reduce the amount of CMake
code it needs to maintain, and brings it inline with the rest
of RAPIDS.

### CODE REVIEW

**Comment by robertmaynard** [comment]
rerun tests

**Comment by robertmaynard** [comment]
rerun tests

**Comment by robertmaynard** [comment]
rerun tests

**Comment by robertmaynard** [comment]
rerun tests

**Review by jlowe — APPROVED** [review]
This seems OK from the Java perspective.  Was able to build this in the Docker container used for publishing cudf Java jar snapshots.

**Comment by devavret** [comment]
I don't believe my review is needed.

File: cpp/cmake/thirdparty/get_libcudacxx.cmake [hunk 1]
```diff
@@ -15,16 +15,14 @@
 #=============================================================================
```
Comment (harrism) [inline cpp/cmake/thirdparty/get_libcudacxx.cmake:15]: Update copyright.

File: cpp/libcudf_kafka/CMakeLists.txt [hunk 2]
```diff
@@ -13,10 +13,23 @@
 # See the License for the specific language governing permissions and
```
Comment (harrism) [inline cpp/libcudf_kafka/CMakeLists.txt:13]: Copyright

File: java/src/main/native/cmake/Modules/ConfigureNvcomp.cmake [hunk 3]
```diff
@@ -35,31 +35,31 @@ execute_process(COMMAND ${CMAKE_COMMAND} -G ${CMAKE_GENERATOR} .
```
Comment (harrism) [inline java/src/main/native/cmake/Modules/ConfigureNvcomp.cmake:35]: Copyright

**Review by harrism — CHANGES_REQUESTED** [review]
Looks great. Just a few copyrights need to be updated.

**Comment by hyperbolic2346** [comment]
I don't feel that I can review this well.

**Comment by robertmaynard** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

**Comment by robertmaynard** [comment]
@gpucibot merge

File: java/src/main/native/CMakeLists.txt [hunk 4]
```diff
@@ -216,67 +189,63 @@ if(NVCOMP_FOUND)
...
-                    "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
-                    "${NVCOMP_INCLUDE_DIR}"
-                    "${CMAKE_BINARY_DIR}/include"
-                    "${CMAKE_SOURCE_DIR}/include"
-                    "${SPDLOG_INCLUDE}"
-                    "${CMAKE_SOURCE_DIR}/src"
-                    "${JNI_INCLUDE_DIRS}"
-                    "${CUDF_INCLUDE}"
-                    "${RMM_INCLUDE}"
-                    "${ARROW_INCLUDE}")
```
Comment (pxLi) [inline java/src/main/native/CMakeLists.txt:246]: removing this part breaks cufilejni build https://github.com/rapidsai/cudf/issues/9167
submitted a fix at #9168

Comment (robertmaynard) [reply java/src/main/native/CMakeLists.txt:246]: Thanks for fixing that regression, sorry I introduced it

Comment (pxLi) [reply java/src/main/native/CMakeLists.txt:246]: no worries, thanks

**Pull request merged**